server/events/working_dir.go

```diff
@@ -290,6 +333,64 @@ func (w *FileWorkspace) forceClone(log logging.Logger,
 	return nil
 }
 
+func (w *FileWorkspace) forceCloneSha(log logging.Logger, cloneDir string, baseRepo models.Repo, sha string) error {
```

Comment:
Eventually we'd probably want to do something like this:
https://sourcegraph.lyft.net/github.com/lyft/orchestration@cfa5491fdafaef9d022e1072b75557d52fb5cc5f/-/blob/nublar/core/fetcher.go?L71

It just gets an archive for a given sha from github, unzips it and then operates within. Seems a lot simpler and straightforward than all these commands. Makes sense to not change this whole process right now though, just wanted to call it out.